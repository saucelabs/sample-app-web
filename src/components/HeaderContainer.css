.header_container {
  display: flex;
  flex-direction: column;
}

.primary_header {
  height: 70px;
}

.primary_header .header_label {
  background: #fff;
  padding: 18px 18px 0;
  text-align: center;
  font-size: 24px;
  position: relative;
}

.primary_header .app_logo {
  background: url("../assets/img/SwagLabs_logo.png") no-repeat center center;
  background: url("../assets/svg/logo3x.svg") no-repeat center center,
    linear-gradient(transparent, transparent);
  text-align: center;
  margin: 0 50px;
  height: 35px;
}

.primary_header .shopping_cart_container {
  height: 40px;
  width: 40px;
  position: absolute;
  top: 20px;
  right: 20px;
}

.header_secondary_container {
  background: #474c55;
  width: 100%;
  height: 80px;
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 0 30px;
  box-sizing: border-box;
}

.header_secondary_container .title {
  color: #fff;
  font-size: 24px;
  text-transform: uppercase;
}

.header_secondary_container .peek {
  background: url("../assets/img/headerBot2x.png") no-repeat center center;
  background: url("../assets/svg/headerBot3x.svg") no-repeat center center,
    linear-gradient(transparent, transparent);
  display: block;
  height: 91px;
  width: 90px;
  margin-top: 13px;
  margin-left: 25px;
}

.right_component {
  margin-left: auto;
}

@media only screen and (max-width: 640px) {
  .header_secondary_container {
    height: 60px;
    padding: 0 15px;
  }

  .header_secondary_container .title {
    font-size: 20px;
  }

  .header_secondary_container .peek {
    display: none;
  }
}
