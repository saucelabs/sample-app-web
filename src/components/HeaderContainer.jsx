import React from "react";
import PropTypes from "prop-types";
import "./HeaderContainer.css";
import DrawerMenu from "./DrawerMenu";
import CartButton from "./CartButton";

const HeaderContainer = ({
  customClass,
  secondaryHeaderBot,
  secondaryLeftComponent,
  secondaryRightComponent,
  secondaryTitle,
}) => {
  /**
   * @TODO:
   * This can't be tested yet because enzyme currently doesn't support ReactJS17,
   * see https://github.com/enzymejs/enzyme/issues/2429.
   * This means we can't fully mount the component and test all rendered components
   * and functions
   */
  /* istanbul ignore next */
  const HeaderBot = () => <div className="peek" />;
  /**
   * @TODO:
   * This can't be tested yet because enzyme currently doesn't support ReactJS17,
   * see https://github.com/enzymejs/enzyme/issues/2429.
   * This means we can't fully mount the component and test all rendered components
   * and functions
   */
  /* istanbul ignore next */
  const LeftComponent = ({ leftComponent }) => (
    <div className="left_component">{React.cloneElement(leftComponent)}</div>
  );
  /**
   * @TODO:
   * This can't be tested yet because enzyme currently doesn't support ReactJS17,
   * see https://github.com/enzymejs/enzyme/issues/2429.
   * This means we can't fully mount the component and test all rendered components
   * and functions
   */
  /* istanbul ignore next */
  const RightComponent = ({ rightComponent }) => (
    <div className="right_component">{React.cloneElement(rightComponent)}</div>
  );
  /**
   * @TODO:
   * This can't be tested yet because enzyme currently doesn't support ReactJS17,
   * see https://github.com/enzymejs/enzyme/issues/2429.
   * This means we can't fully mount the component and test all rendered components
   * and functions
   */
  /* istanbul ignore next */
  const Title = ({ title }) => <span className="title">{title}</span>;
  const extraClass = customClass ? ` ${customClass}` : "";

  return (
    <div id="header_container" className={`header_container${extraClass}`}>
      <div className="primary_header">
        <div id="menu_button_container">
          <DrawerMenu />
        </div>
        <div className="header_label">
          <div className="app_logo" />
        </div>
        <div id="shopping_cart_container" className="shopping_cart_container">
          <CartButton />
        </div>
      </div>
      <div className="header_secondary_container">
        {secondaryLeftComponent && (
          <LeftComponent leftComponent={secondaryLeftComponent} />
        )}
        {secondaryTitle && <Title title={secondaryTitle} />}
        {secondaryHeaderBot && <HeaderBot />}
        {secondaryRightComponent && (
          <RightComponent rightComponent={secondaryRightComponent} />
        )}
      </div>
    </div>
  );
};
HeaderContainer.propTypes = {
  /**
   * A custom class for the header container
   */
  customClass: PropTypes.string,
  /**
   * Show the header bot
   */
  secondaryHeaderBot: PropTypes.bool,
  /**
   * A react component for the left
   */
  secondaryLeftComponent: PropTypes.element,
  /**
   * A react component for the right
   */
  secondaryRightComponent: PropTypes.element,
  /**
   * A title
   */
  title: PropTypes.string,
};
HeaderContainer.defaultProps = {
  customClass: undefined,
  secondaryHeaderBot: undefined,
  secondaryLeftComponent: undefined,
  secondaryRightComponent: undefined,
  secondaryTitle: undefined,
};

export default HeaderContainer;
