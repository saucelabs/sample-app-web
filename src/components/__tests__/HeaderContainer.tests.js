import React from "react";
import { shallow } from "enzyme";
import HeaderContainer from "../HeaderContainer";

describe("HeaderContainer", () => {
  it("should render without any props", () => {
    const component = shallow(<HeaderContainer />);

    expect(component).toMatchSnapshot();
  });

  it("should render with all props", () => {
    const component = shallow(
      <HeaderContainer
        customClass="custom_class"
        secondaryHeaderBot
        secondaryLeftComponent={<input type="checkbox" />}
        secondaryRightComponent={<input type="radio" />}
        secondaryTitle="sample title"
      />
    );

    expect(component).toMatchSnapshot();
  });
});
