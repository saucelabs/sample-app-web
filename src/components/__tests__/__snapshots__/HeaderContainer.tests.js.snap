// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer should render with all props 1`] = `
<div
  className="header_container custom_class"
  id="header_container"
>
  <div
    className="primary_header"
  >
    <div
      id="menu_button_container"
    >
      <withRouter(DrawerMenu) />
    </div>
    <div
      className="header_label"
    >
      <div
        className="app_logo"
      />
    </div>
    <div
      className="shopping_cart_container"
      id="shopping_cart_container"
    >
      <withRouter(CartButton) />
    </div>
  </div>
  <div
    className="header_secondary_container"
  >
    <LeftComponent
      leftComponent={
        <input
          type="checkbox"
        />
      }
    />
    <Title
      title="sample title"
    />
    <HeaderBot />
    <RightComponent
      rightComponent={
        <input
          type="radio"
        />
      }
    />
  </div>
</div>
`;

exports[`HeaderContainer should render without any props 1`] = `
<div
  className="header_container"
  id="header_container"
>
  <div
    className="primary_header"
  >
    <div
      id="menu_button_container"
    >
      <withRouter(DrawerMenu) />
    </div>
    <div
      className="header_label"
    >
      <div
        className="app_logo"
      />
    </div>
    <div
      className="shopping_cart_container"
      id="shopping_cart_container"
    >
      <withRouter(CartButton) />
    </div>
  </div>
  <div
    className="header_secondary_container"
  />
</div>
`;
